### PR TITLE
[gitpod-db] Include redis client to share WEB-621

### PR DESCRIPTION
**Pull request by easyCZ**
## Description

Push the Redis client into `gitpod-db` such that it can be reused between server and ws-man-bridge. The reids publisher code will follow after this PR

## Related Issue(s)

Fixes #

#### Preview status

<p>Gitpod was successfully deployed to your preview environment.</p>
<ul>
	<li><b>🏷️ Name</b> - mp-db-inclf84ecf3c66</li>
	<li><b>🔗 URL</b> - <a href="https://mp-db-inclf84ecf3c66.preview.gitpod-dev.com/workspaces" target="_blank">mp-db-inclf84ecf3c66.preview.gitpod-dev.com/workspaces</a>.</li>
	<li><b>📚 Documentation</b> - See our <a href="https://www.notion.so/gitpod/6debd359591b43688b52f76329d04010#7c1ce80ab31a41e29eff2735e38eec39" target="_blank">internal documentation</a> for information on how to interact with your preview environment.</li>
	<li><b>📦 Version</b> - mp-db-include-redis-client-gha.13472</li>
</ul>

## Build Options

<details>
<summary>Build</summary>

- [ ] /werft with-werft
      Run the build with werft instead of GHA
- [ ] leeway-no-cache
- [ ] /werft no-test
      Run Leeway with `--dont-test`
</details>

<details>
<summary>Publish</summary>

- [ ] /werft publish-to-npm
- [ ] /werft publish-to-jb-marketplace
</details>

<details>
<summary>Installer</summary>

- [ ] analytics=segment
- [ ] with-dedicated-emulation
- [ ] workspace-feature-flags
  Add desired feature flags to the end of the line above, space separated
</details>

<details>
<summary>Preview Environment</summary>

- [ ] /werft with-local-preview
      If enabled this will build `install/preview`
- [x] /werft with-preview
- [ ] /werft with-large-vm
- [ ] /werft with-gce-vm
      If enabled this will create the environment on GCE infra
- [ ] with-integration-tests=all
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`
</details>

/hold

### CODE REVIEW

File: components/ws-manager-bridge/src/redis/publisher.ts [hunk 1]
```diff
@@ -16,21 +15,19 @@ import {
     RedisWorkspaceInstanceUpdate,
     WorkspaceInstanceUpdatesChannel,
 } from "@gitpod/gitpod-protocol";
+import { Redis } from "ioredis";
 
 @injectable()
 export class RedisPublisher {
```
Comment (svenefftinge) [inline components/ws-manager-bridge/src/redis/publisher.ts:21]: Is not copying the publisher in two places the main problem?

Comment (easyCZ) [reply components/ws-manager-bridge/src/redis/publisher.ts:21]: It is, but I'm first moving the Redis Client, and then moving the publisher. To move the publisher, I'll also need to add prometheus metrics to do the `gitpod-db` package which will be a bigger change so trying to slice it up

**Comment by geropl** [comment]
Build seemed to got stuck, so cancelled it: https://github.com/gitpod-io/gitpod/actions/runs/5642731312/job/15283546638?pr=18264

**Review by geropl — APPROVED** [review]
Code LGTM, tested and works :heavy_check_mark:

**Pull request merged**